WcaOnRails/app/models/team.rb

```diff
@@ -219,4 +219,136 @@ def serializable_hash(options = nil)
+      new_member = false
+      ex_leader = false
+      ex_senior_member = false
+      ex_member = false
+      new_and_ex_leader = false
+      new_and_ex_senior_member = false
+      new_and_ex_member = false
+
+      # Creates flags for what all actioned in the duration.
+      team_member_changes[user.id].each do |member|
+        if member.end_date && member.end_date >= duration_start_date
+          if member.start_date >= duration_start_date
+            new_and_ex_leader = member.team_leader
+            new_and_ex_senior_member = member.team_senior_member
+            new_and_ex_member = !new_and_ex_leader && !new_and_ex_senior_member
+          else
+            ex_leader = member.team_leader
+            ex_senior_member = member.team_senior_member
+            ex_member = !ex_leader && !ex_senior_member
+          end
+        elsif !member.end_date && member.start_date >= duration_start_date
+          new_leader = member.team_leader
+          new_senior_member = member.team_senior_member
+          new_member = !new_leader && !new_senior_member
+        end
+      end
+
+      # Gets assigned to respective group of changes
+      name = user.name
+      if new_leader || ex_leader || new_and_ex_leader
+        if new_leader
+          leader_changes.append("#{name} has been appointed as the new Leader.")
+        elsif ex_leader
+          if new_member
+            leader_changes.append("#{name} is no longer the Leader, but will continue as member.")
+          elsif new_senior_member
+            leader_changes.append("#{name} is no longer the Leader, but will continue as Senior member.")
+          else
+            leader_changes.append("#{name} is no longer the Leader and no longer a member.")
+          end
+        else
+          if new_member
+            leader_changes.append("#{name} was the Leader for few days and is continuing as member.")
+          elsif new_senior_member
+            leader_changes.append("#{name} was the Leader for few days and is continuing as Senior member.")
+          else
+            leader_changes.append("#{name} was the Leader for few days and is no longer a member.")
+          end
+        end
+      else
+        if new_senior_member || new_and_ex_senior_member
+          if ex_member || new_and_ex_member
+            promoted_senior_members.append(name)
+          else
+            new_senior_members.append(name)
+          end
+        end
+        if new_member || new_and_ex_member
+          if !ex_senior_member && !new_and_ex_senior_member
+            new_members.append(name)
+          else
+            demoted_senior_members.append(name)
+          end
+        elsif ex_senior_member || new_and_ex_senior_member
+          no_more_senior_members.append(name)
+        end
+        if (ex_member || new_and_ex_member) && !new_senior_member && !new_member
+          no_more_members.append(name)
+        end
+      end
+    end
+
+    changes_of_last_month = ""
```

Comment:
Can you build this as a list instead of a string, and then `.join('\n')` at the very end?